test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/BulkUpdateSaveResultTests.cs

```diff
@@ -0,0 +1,288 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="BulkUpdateSaveResultTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.OData.Client.TDDUnitTests.Tests
+{
+    /// <summary>
+    /// Unit tests for the BulkUpdateSaveResult class.
+    /// </summary>
+    public class BulkUpdateSaveResultTests
+    {
+        private DataServiceContext context;
+
+        public BulkUpdateSaveResultTests()
+        {
+            context = new DataServiceContext(new Uri("http://www.odata.org/service.svc")).ReConfigureForNetworkLoadingTests();
+        }
+
+        [Fact]
+        public void SerealizeADeltaEntry_With_TwoNavigationLinks()
+        {
+            var person = new Person
+            {
+                ID = 100,
+                Name = "Bing",
+            };
+
+            var car1 = new Car { ID = 1001 };
+            var car2 = new Car { ID = 1002 };
+
+
+            context.AttachTo("Persons", person);
+            context.AttachTo("Cars", car1);
+            context.AttachTo("Cars", car2);
+            context.AddLink(person, "Cars", car1);
+            context.AddLink(person, "Cars", car2);
+
+            var requestInfo = new RequestInfo(context);
+            var serializer = new Serializer(requestInfo);
+            var headers = new HeaderCollection();
+            var clientModel = new ClientEdmModel(ODataProtocolVersion.V4);
+            var entityDescriptor = new EntityDescriptor(clientModel);
+            entityDescriptor.State = EntityStates.Added;
+            entityDescriptor.EntitySetName = "People";
+            entityDescriptor.Entity = person;
+            entityDescriptor.EditLink = new Uri("http://www.foo.com/custom");
+            var requestMessageArgs = new BuildingRequestEventArgs("PATCH", new Uri("http://www.foo.com/Northwind/$metadata#People/$delta"), headers, entityDescriptor, HttpStack.Auto);
+            var odataRequestMessageWrapper = ODataRequestMessageWrapper.CreateRequestMessageWrapper(requestMessageArgs, requestInfo);
+            entityDescriptor.AddRelatedDescriptor(new LinkDescriptor(person, "Cars", car1, clientModel));
+            entityDescriptor.AddRelatedDescriptor(new LinkDescriptor(person, "Cars", car2, clientModel));
+            List<Descriptor> entries = new List<Descriptor> { entityDescriptor };

```

Comment:
I assume the customer would call `dsc.BulkUpdate()` method. I also assume the customer would not have to manually create and modify descriptors. For that reason, I think you should also have tests that mimick how the customer would interact with the feature.

Reply:
I wanted to add these end to end tests with the response deserialization changes.